risc0/zkvm/src/prove/mod.rs

```diff
@@ -127,6 +145,40 @@ impl<'a> Default for ProverOpts<'a> {
 }
 
 /// Manages communication with and execution of a zkVM [Program]
+///
+/// # Usage
+/// A [Prover] is constructed from the ELF code and an Image ID generated from
+/// the guest code to be proven (see
+/// [risc0_build](https://docs.rs/risc0-build/latest/risc0_build/) for more
+/// information about how these are generated). Use [Prover::new] if you want
+/// the default [ProverOpts], or [Prover::new_with_opts] to use custom options.
+/// ```ignore
+/// // In real code, the ELF & Image ID would be generated by risc0 build scripts from guest code
+/// use methods::{EXAMPLE_ELF, EXAMPLE_ID};
+/// use risc0_zkvm::Prover;
+///
+/// let mut prover = Prover::new(&EXAMPLE_ELF, EXAMPLE_ID)?;
+/// ```
+/// Provers should essentially always be mutable so that their [Prover::run]
+/// method may be called.
+///
+/// Input data can be passed to the Prover with [Prover::add_input_u32_slice]
+/// (or [Prover::add_input_u8_slice]). After all inputs have been added, call
+/// [Prover::run] to execute the guest code and produce a [Receipt] proving
+/// execution.
+/// ```ignore
+/// prover.add_input_u32_slice(&to_vec(&input)?);
```

Comment:
I don't think this compiles, because `to_vec` isn't qualified and there's no corresponding `use`.

Reply:
Hmm, I think I'm missing something because a very similar line works fine in the factors example:
https://github.com/risc0/risc0/blob/861c3959d5a06bb1826a31cf0e5345d97251ec58/examples/factors/src/main.rs#L33
Is it that in factors the `a` is a `u64`, and if `input` were a more complex type there might be ambiguity that doesn't exist with a `u64`?

Reply:
The factors example has `use risc0_zkvm::serde::to_vec`. I'm saying the example either needs to have a `use` or it needs to fully qualify `to_vec`.

Reply:
gotcha, thanks!